lib/processors/versionInfoGenerator.js

```diff
@@ -14,6 +16,337 @@ function getTimestamp() {
+							resolvedLib.lazy || dependencyInfoObjectAdded.lazy);
+					}
+				} else {
+					log.info(`Cannot find dependency '${libName}' `+
+						`defined in the manifest.json or .library file of project '${this.name}'. ` +
+						"This might prevent some UI5 runtime performance optimizations from taking effect. " +
+						"Please double check your project's dependency configuration.");
+				}
+			}
+		}
+		return this._libsResolved;
+	}
+}
+
+
+/**
+ * Sorts the keys of a given object
+ *
+ * @param {object} obj the object
+ * @returns {object} the object with sorted keys
+ */
+const sortObjectKeys = (obj) => {
+	const sortedObject = {};
+	const keys = Object.keys(obj);
+	keys.sort();
+	keys.forEach((key) => {
+		sortedObject[key] = obj[key];
+	});
+	return sortedObject;
+};
+
+/**
+ * Builds the manifestHints object from the dependencyInfo
+ *
+ * @param {DependencyInfo} dependencyInfo
+ * @param {Map<string, DependencyInfo>} dependencyInfoMap
+ * @returns {{dependencies: {libs: ManifestLibraries}}} manifestHints
+ */
+const getManifestHints = (dependencyInfo, dependencyInfoMap) => {
+	if (dependencyInfo) {
+		const libsResolved = dependencyInfo.getResolvedLibraries(dependencyInfoMap);
+		if (libsResolved && Object.keys(libsResolved).length) {
+			return {
+				dependencies: {
+					libs: sortObjectKeys(libsResolved)
+				}
+			};
+		}
+	}
+};
+
+/**
+ * Common type for Library and Component
+ * embeds and bundled components makes only sense for library
+ *
+ * @typedef {object} ArtifactInfo
+ * @property {string} componentName The library name, e.g. "lib.x"
+ * @property {Set<string>} bundledComponents The embedded components which have a embeddedBy reference to the library
+ * @property {DependencyInfo} dependencyInfo The dependency info object
+ * @property {ArtifactInfo[]} embeds The embedded artifact infos
+ */
+
+
+/**
+ * Processes the manifest and creates a ManifestInfo and an ArtifactInfo.
+ *
+ * @param {module:@ui5/fs.Resource} libraryManifest
+ * @param {string} [name] library name, if not provided using the ManifestInfo's id
+ * @returns {Promise<{manifestInfo: ManifestInfo, libraryArtifactInfo: ArtifactInfo}>}
+ */
+async function processManifestAndGetArtifactInfo(libraryManifest, name) {
+	const manifestInfo = await processManifest(libraryManifest);
+	name = name || manifestInfo.id;
+	const libraryArtifactInfo = Object.create(null);
+	libraryArtifactInfo.componentName = name;
+	libraryArtifactInfo.dependencyInfo = new DependencyInfo(manifestInfo.libs, name);
+	return {manifestInfo, libraryArtifactInfo};
+}
+
+/**
+ * Processes the library info and fills the maps <code>dependencyInfoMap</code> and <code>embeddedInfoMap</code>.
+ *
+ * @param {LibraryInfo} libraryInfo
+ * @returns {Promise<ArtifactInfo|undefined>}
+ */
+const processLibraryInfo = async (libraryInfo) => {
+	if (!libraryInfo.libraryManifest) {
+		log.warn(
+			`Cannot add meta information for library '${libraryInfo.name}'. The manifest.json file cannot be found`);
```

Comment:
When executing a `ui5 build` without `--all`, this warning appears for all framework dependencies since no manifest.json files are generated for them.

I guess we already discussed that this task doesn't really make sense when executed without `--all` (hence the new comment in builder.js). But since it is executed in such builds as of today, I find the warning irritating and think it should not appear in that case.

Maybe we need to check whether the has not been supplied at all?

Example from the openui5-sample-app:
```
> ui5 build
info normalizer:translators:ui5Framework Using OpenUI5 version: 1.85.2
info builder:builder Building project openui5-sample-app not including dependencies...
info builder:builder 🛠  (1/1) Building project openui5-sample-app
info builder:builder application openui5-sample-app 🔨 (1/8) Running task escapeNonAsciiCharacters...
info builder:builder application openui5-sample-app 🔨 (2/8) Running task replaceCopyright...
info builder:builder application openui5-sample-app 🔨 (3/8) Running task replaceVersion...
info builder:builder application openui5-sample-app 🔨 (4/8) Running task generateFlexChangesBundle...
info builder:builder application openui5-sample-app 🔨 (5/8) Running task generateComponentPreload...
info builder:builder application openui5-sample-app 🔨 (6/8) Running task createDebugFiles...
info builder:builder application openui5-sample-app 🔨 (7/8) Running task uglify...
info builder:builder application openui5-sample-app 🔨 (8/8) Running task generateVersionInfo...
WARN builder:processors:versionInfoGenerator Cannot add meta information for library 'sap.ui.unified'. The manifest.json file cannot be found
WARN builder:processors:versionInfoGenerator Cannot add meta information for library 'sap.m'. The manifest.json file cannot be found
WARN builder:processors:versionInfoGenerator Cannot add meta information for library 'sap.ui.core'. The manifest.json file cannot be found
WARN builder:processors:versionInfoGenerator Cannot add meta information for library 'sap.ui.layout'. The manifest.json file cannot be found
WARN builder:processors:versionInfoGenerator Cannot add meta information for library 'sap.f'. The manifest.json file cannot be found
info builder:builder Build succeeded in 574 ms
info builder:builder Executing cleanup tasks...
```

Reply:
tackled with 42ce003
the log is verbose now and does not occur anymore when building the openui5 sample app without `--all`